tests/security/389ds/tls_389ds_sssd_client.pm

```diff
@@ -0,0 +1,76 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Implement & Integrate 389ds + sssd test case into openQA,
+#          This test module covers the sssd client tests
+#
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#88513, tc#1768672
+
+use base 'opensusebasetest';
+use base 'consoletest';
+use testapi;
+use strict;
+use warnings;
+use utils;
+use lockapi;
+use Utils::Systemd qw(systemctl disable_and_stop_service);
+use version_utils 'is_tumbleweed';
+
+sub run {
+    select_console("root-console");
+
+    my $remote_ip   = '10.0.2.101';
+    my $remote_name = '389ds';
+    my $inst_ca_dir = '/etc/dirsrv/slapd-localhost';
+    my $tls_dir     = '/etc/openldap/certs';
+    my $ldap_user   = $testapi::username;
+    my $uid         = '1003';
+
+    # Install 389-ds and sssd on client
+    zypper_call("in 389-ds sssd");
+
+    # Disable and stop the nscd daemon because it conflicts with sssd
+    disable_and_stop_service("nscd", ignore_failure => 1);
+
+    # Copy the /etc/hosts, sample sssd.conf and CA files from server
+    # Configure tls CA key on client
+    mutex_wait("389DS_READY");
+    assert_script_run("mkdir -p $tls_dir");
+    exec_and_insert_password("scp -o StrictHostKeyChecking=no root\@$remote_ip:/etc/hosts /etc/hosts");
+    exec_and_insert_password("scp -o StrictHostKeyChecking=no root\@$remote_ip:/tmp/sssd.conf /tmp");
+    exec_and_insert_password("scp -o StrictHostKeyChecking=no root\@$remote_ip:$inst_ca_dir/ca.crt $tls_dir");
+    assert_script_run("/usr/bin/openssl rehash $tls_dir");
+
+    # For Tumbleweed, the nsswitch.conf file locates at /usr/etc
+    assert_script_run("cp /usr/etc/nsswitch.conf /etc") if is_tumbleweed;
```

Comment:
Our tests should always consider openSUSE Tumbleweed as the default product without a version. That means any differing behaviour should have an explicit exclude rule for the "older products" and potentially an explanation why the behaviour should differ. You can try to negate the logic check.

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

Reply:
@okurz I used the method #1, can you please double check?